webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/TableRow.js

```diff
@@ -0,0 +1,287 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Tr, Td, TableText } from '@patternfly/react-table';
+import {
+  TimesIcon,
+  CheckIcon,
+  PencilAltIcon,
+  FlagIcon,
+} from '@patternfly/react-icons';
+import {
+  SelectVariant,
+  Select,
+  SelectOption,
+  Button,
+  TextInput,
+  TextArea,
+  Checkbox,
+  FormGroup,
+  Form,
+  Tooltip,
+} from '@patternfly/react-core';
+import { APIActions } from '../../../../redux/API';
+import { sprintf, translate as __ } from '../../../../common/I18n';
+import { HOST_PARAM, columnNames, typeOptions } from './ParametersConstants';
+import { updateHost } from '../../ActionsBar/actions';
+import { RowActions } from './RowActions';
+
+export const ParametersTableRow = ({
+  param,
+  rowIndex,
+  editingRow,
+  setEditingRow,
+  hostId,
+  editHostsPermission,
+  isNew,
+}) => {
+  const [name, setName] = useState(param.name);
+  const [type, setType] = useState(param.parameter_type);
+  const [value, setValue] = useState(param.value);
+  const [selectIsOpen, setSelectIsOpen] = useState(false);
+  const [selectValueIsOpen, setSelectValueIsOpen] = useState(false);
+  const [isHide, setIsHide] = useState(param['hidden_value?']);
+
+  const dispatch = useDispatch();
+  const getValue = () => {
+    if (param['hidden_value?']) {
+      return '*****';
+    }
+    if (param.parameter_type === 'boolean') {
+      return param.value.toString();
+    }
+    if (!param.value)
+      return <span className="disabled-text">{__('No value')}</span>;
+    if (['json', 'yaml', 'array', 'hash'].includes(param.parameter_type)) {
+      return JSON.stringify(param.value);
+    }
+    return param.value;
+  };
+  const APIActionsProps = {
+    errorToast: ({ response }) =>
+      // eslint-disable-next-line camelcase
+      response?.data?.error?.full_messages || response?.data?.error?.message,
+    handleSuccess: () => {
+      dispatch(updateHost(hostId));
+    },
+  };
+  const onSubmit = () => {
+    if (isNew || param.associated_type !== HOST_PARAM) {
+      dispatch(
+        APIActions.post({
```

Comment:
Right now it's not too bad since there's only a couple actions, but if it gets much bigger than this it may be better to put the action creators in a different file and then import them.

Reply:
If this file will get more than 300 lines (and make lint sad) in the next pr I'll move them